### PR TITLE
GHA updates

### PR DESCRIPTION
**Pull request by timcosgrove**
## Description

This is a re-post of #1554 

relates to https://github.com/department-of-veterans-affairs/va.gov-cms/issues/12640

This finishes up some remaining work to update content-build across the board for the GHA changes. Changes:
* Moves https://github.com/gregziegan/fetch-latest-release into our repo, as it was only used in one place and the original is not being maintained
* Sets up Dependabot for this repo, including for Github Actions
* Makes all third party action references hashes, per Github's recommendations; this also assists Dependabot in keeping them up-to-date

## Testing done & Screenshots
With the previous merge, problems appeared to be with the configure-aws-credentials action. Here are successful test runs of that action, from this PR's triggered actions:
https://github.com/department-of-veterans-affairs/content-build/actions/runs/4983302804
https://github.com/department-of-veterans-affairs/content-build/actions/runs/4983302806/jobs/8920143851
https://github.com/department-of-veterans-affairs/content-build/actions/runs/4983401614/jobs/8920374992#step:2:1

## Acceptance criteria

- [ ] After merge, all actions continue to function correctly.

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    time: "09:00"
+    timezone: "America/New_York"
+  # Allow up to 10 open pull requests for npm dependencies
+  open-pull-requests-limit: 0
+  ignore:
+    # Ignore updates to packages that start with 'cypress'
+    # Wildcards match zero or more arbitrary characters
+    - dependency-name: "cypress*"
+    - dependency-name: "@testing-library/*"
+    - dependency-name: "eslint*"
```
Comment (ndouglas) [inline .github/dependabot.yml:23]: bold move

**Review by ndouglas — COMMENTED** [review]
This is great stuff. Pretty eye-watering amount of work.

File: .github/workflows/preview-environment-deployment.yml.bak
Comment (ndouglas) [inline .github/workflows/preview-environment-deployment.yml.bak:1]: Is this file necessary?

File: src/site/stages/build/drupal/individual-queries.js
Comment (ndouglas) [inline src/site/stages/build/drupal/individual-queries.js:1]: Is this change meant to be part of this PR?

File: .github/dependabot.yml (hunk 1 above)
Comment (timcosgrove) [reply .github/dependabot.yml:23]: In a bad way? Or just, adding NPM updates?

File: .github/workflows/preview-environment-deployment.yml.bak
Comment (timcosgrove) [reply .github/workflows/preview-environment-deployment.yml.bak:1]: It is not mine and I do not want to remove it on them; however, I want to make it no longer recognized as a workflow / .yml file

File: src/site/stages/build/drupal/individual-queries.js
Comment (timcosgrove) [reply src/site/stages/build/drupal/individual-queries.js:1]: No, thank you for catching; bad merge on my part. I will clean.

File: .github/dependabot.yml (hunk 1 above)
Comment (ndouglas) [reply .github/dependabot.yml:23]: Sorry, I was semi-joking. It's a good change and I should've marked that I was making a light-hearted comment.

File: .github/workflows/preview-environment-deployment.yml.bak
Comment (ndouglas) [reply .github/workflows/preview-environment-deployment.yml.bak:1]: Gotcha.

File: .github/dependabot.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+version: 2
+updates:
+- package-ecosystem: "github-actions"
+  directory: "/"
+  schedule:
+    # Check for updates to GitHub Actions every week
+    interval: "weekly"
+# This npm dependabot config is lifted from https://github.com/department-of-veterans-affairs/vets-website/blob/main/.github/dependabot.yml.
+# We will differentiate and refine over time.
+- package-ecosystem: "npm"
```
Comment (ndouglas) [inline .github/dependabot.yml:10]: Another possibly dumb question: this refers to `npm` and the project uses `yarn`, but this is still appropriate because it's referring to the ecosystem and not the specifics of the tools, right?

**Review by ndouglas — APPROVED** [review]
Looks great!

**Review by olivereri — APPROVED** [review]

File: .github/dependabot.yml (hunk 2 above)
Comment (timcosgrove) [reply .github/dependabot.yml:10]: Not a dumb question. I assumed it would be the same, but that appears to be wrong. I'll shift it, just a sec.

**Comment by timcosgrove** [comment]
> This is great stuff. Pretty eye-watering amount of work.

Was actually not terrible. Caffeine is a helluva drug.
